src/pixelator/report/qcreport/main.py

```diff
@@ -0,0 +1,230 @@
+    :param sample_name: the name of the sample
+    :param info: A dictionary containing meta information about the sample
+    :param output_path: the path to the output folder
+    """
+    sample_id = info.sample_id
+    logger.info("Creating QC report for sample %s", sample_id)
+
+    amplicon_metrics = reporting.amplicon_metrics(sample_name)
+    preqc_metrics = reporting.preqc_metrics(sample_name)
+    demux_metrics = reporting.demux_metrics(sample_name)
+    collapse_metrics = reporting.collapse_metrics(sample_name)
+    graph_metrics = reporting.graph_metrics(sample_name)
+    annotate_metrics = reporting.annotate_metrics(sample_name)
+    reads_flow = reporting.reads_flow(sample_name)
+    molecules_flow = reporting.molecules_flow(sample_name)
+
+    # Collect antibody metrics
+    antibodies_data_values = {
+        "antibody_reads": demux_metrics.output_read_count,
+        "antibody_reads_usable_per_cell": annotate_metrics.read_count_per_cell_stats.mean,
+        "antibody_reads_in_outliers": annotate_metrics.reads_in_aggregates_count,
+        "unrecognized_antibodies": demux_metrics.unrecognised_antibody_read_count,
+    }
+
+    antibodies_data_fractions = (
+        pd.Series(antibodies_data_values) / reads_flow.input_read_count
+    ).to_dict()
+
+    antibodies_data_fractions = {
+        f"fraction_{k}": v for k, v in antibodies_data_fractions.items()
+    }
+
+    placeholder_cell_predictions = {
+        "predicted_cell_type_b_cells": None,
+        "fraction_predicted_cell_type_b_cells": None,
+        "predicted_cell_type_cd4p_cells": None,
+        "fraction_predicted_cell_type_cd4p_cells": None,
+        "predicted_cell_type_cd8p_cells": None,
+        "fraction_predicted_cell_type_cd8p_cells": None,
+        "predicted_cell_type_monocytes": None,
+        "fraction_predicted_cell_type_monocytes": None,
+        "predicted_cell_type_nk_cells": None,
+        "fraction_predicted_cell_type_nk_cells": None,
+        "predicted_cell_type_unknown": None,
+        "fraction_predicted_cell_type_unknown": None,
+    }
+
+    # The total number of input reads per cell.
+    # Note that this is not the number of actual reads that were assigned to the cell.
+    # This gives an idea of how many reads are lost in the pipeline averaged over
+    # the number of cells
+    average_input_read_per_cell = (
+        reads_flow.input_read_count / annotate_metrics.cell_count
+    )
+
+    # Map pixelator metrics to QC report metrics
+    metrics = Metrics(
+        number_of_cells=annotate_metrics.cell_count,
+        average_reads_usable_per_cell=int(
+            annotate_metrics.read_count_per_cell_stats.mean
+        ),
+        average_reads_per_cell=int(average_input_read_per_cell),
+        average_antibody_molecules_per_cell=annotate_metrics.molecule_count_per_cell_stats.mean,
+        average_upias_per_cell=round(
+            annotate_metrics.b_pixel_count_per_cell_stats.mean
+        ),
+        average_umis_per_upia=round(
```

Comment:
That is still the goal yes. This is done because the qc-report currently expects int inputs.
We should change the qc report to accept floats. Then we can remove these round calls here.